lib/Common/Memory/CollectionState.h

```diff
@@ -91,3 +97,43 @@ enum CollectionState
     CollectionStatePostSweepRedeferralCallback = Collection_PostSweepRedeferralCallback,
     CollectionStatePostCollectionCallback = Collection_PostCollectionCallback,
 };
+
+class CollectionStateChangedObserver
+{
+public:
+    virtual void CollectionStateChanged(CollectionState newVal, CollectionState oldVal) = 0;
+};
+
+class ObservableCollectionState
+{
+private:
+    CollectionState state;
+    CollectionStateChangedObserver* observer;
+    
+    void SetState(CollectionState newState)
```

Comment:
Is it worth deleting the copy constructor from this class and also marking it sealed for now?